Rubberduck.VBEEditor/ReferenceInfo.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor
+{
+    public readonly struct ReferenceInfo
```

Comment:
There is a fair overlap between this struct and [this class](https://github.com/rubberduck-vba/Rubberduck/blob/1cf6e0f9be5d16111323731804bd6feb678887ac/Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeInfos.cs#L149). Should we consider consolidating?

Reply:
I can grab this after you're done if you consider it out of scope. It also overlaps with `COMProject`.

Reply:
I just realized that `TypeInfoReference` cannot implement `IReference` because that would require returning an `IVBE`.

Reply:
My concern is that an API that provides 3 different ways to get same answers does not really make for intuitive and easy to use API. We could consider replacing the fields on both `TypeInfoReference` and `COMProject` with a single `ReferenceInfo` field, which would mean both can be used to provide the "QMN" from different contexts. That may help clarify the differences and standardize the marshaling of information about a reference. On the other hand, if each provide different answers for same reference, we have a problem.